src/systems/fem_context.C

```diff
@@ -1840,7 +1878,15 @@ void FEMContext::pre_fe_reinit(const System & sys, const Elem * e)
                 (sub_dofs, n_dofs_var);
 
               sub_dofs += n_dofs_var;
-            }
+            };
+
+          if (_active_vars)
+            for (auto v : *_active_vars)
+              init_localized_var_data(v);
+          else
+            for (auto v : make_range(sys.n_vars()))
+              init_localized_var_data(v);
+
```

Comment:
Regarding our previous IM discussion, could we have a `make_range` that takes two args (untested pseudo code)
```
template<typename P, typename T>
IntRange<T>
make_range(P * ptr, T end)
{
  if (ptr)
    return {*std::begin(ptr), *std::end(ptr)};
  else
    return {T(0), end};
}
```
You might have to do some template magic to make this compile, but then you'd use it like. 
```
for (auto v : make_range(_active_vars.get(), sys.n_vars()))
  init_localized_var_data(v);
```
It seems a little specific to this use case to be a general utility, but might be good as an anonymous helper function.

Reply:
We can't just return an IntRange here because that does contiguous incrementing.  We'd need a whole new class to return, one where the iterator checks the container pointer at every increment, and increments an internal value in the null case or a container iterator in the non-null case.  I think it'd be more of a pain than the existing code here, though I'd love it if I was wrong.

Reply:
Oh, yeah, duh. Not sure what I was thinking...